lib/Interpreter/TransactionUnloader.cpp

```diff
@@ -634,49 +699,12 @@ namespace clang {
     }
 
     // Cleanup the lookup tables.
-    StoredDeclsMap *Map = DC->getPrimaryContext()->getLookupPtr();
-    if (Map) { // DeclContexts like EnumDecls don't have lookup maps.
-      // Make sure we the decl doesn't exist in the lookup tables.
-      StoredDeclsMap::iterator Pos = Map->find(ND->getDeclName());
-      if ( Pos != Map->end()) {
-        // Most decls only have one entry in their list, special case it.
-        if (Pos->second.getAsDecl() == ND)
-          Pos->second.remove(ND);
-        else if (StoredDeclsList::DeclsTy* Vec = Pos->second.getAsVector()) {
-          // Otherwise iterate over the list with entries with the same name.
-          for (StoredDeclsList::DeclsTy::const_iterator I = Vec->begin(),
-                 E = Vec->end(); I != E; ++I)
-            if (*I == ND)
-              Pos->second.remove(ND);
-        }
-        if (Pos->second.isNull() ||
-            (Pos->second.getAsVector() && !Pos->second.getAsVector()->size()))
-          Map->erase(Pos);
-      }
+    // DeclContexts like EnumDecls don't have lookup maps.
+    if (StoredDeclsMap *Map = DC->getPrimaryContext()->getLookupPtr()) {
+      eraseDeclFromMap(Map, ND);
+      checkDeclIsGone(Map, ND);  // Only #ifndef NDEBUG
```

Comment:
Can you wrap the function invocation here in #ifndef NDEBUG and remove it from checkDeclIsGone's definition. This makes it more clear and saves one call.